test/static/utf8/utf8.tests.ps1

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'Verify UTF-8' {
+    BeforeAll {
+        Push-Location $PSScriptRoot\..\..\..
+        $nonBinaryFiles = (git grep -I --files-with-matches -e $'')
+    }
+
+    AfterAll {
+        Pop-Location
+    }
+
+    function hasBom {
+        $bytes = [byte[]]::new(3)
+        $stream = [System.IO.File]::OpenRead($_)
+        $stream.Read($bytes, 0, 3) > $null
+        $stream.Close()
+        return $bytes[0] -eq 0xef -and $bytes[1] -eq 0xbb -and $bytes[2] -eq 0xbf
+    }
+
+    $nonBinaryFiles.ForEach{
+        Context "'$_'" {
+            It 'file has no BOM' {
```

Comment:
Why do we need Context?

Reply:
we could add additional tests, e.g. detect unicode substition of space with nbsp (prevent regression from #12576)

Reply:
I see your point to group tests by file.
```suggestion
        Context "File: $_" {
            It 'No BOM' {
```

Reply:
I copied the structure from `markdown-link.tests.ps1` but I'm not so sure if we need a seperate test for each file. There all 2368 files and there is an overhead.

Reply:
We could report only errors.